### PR TITLE
LPS-87353 Do not reformat the number when building context or saving forms answer

### PR DESCRIPTION
**Pull request by gabrielwas**

### CODE REVIEW

**Comment by gabrielwas** [comment]
With modifications suggested on #497

**Comment by gabrielwas** [comment]
ci:test:sf

**Comment by gabrielwas** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMFormValuesMergerImpl.java [hunk 1]
```diff
@@ -69,10 +76,28 @@ protected DDMFormFieldValue getDDMFormFieldValueByName(
...
 					existingDDMFormFieldValues, newDDMFormFieldValue.getName());
 
+			DDMFormValues ddmFormValues =
+				newDDMFormFieldValue.getDDMFormValues();
+
+			DDMForm ddmForm = ddmFormValues.getDDMForm();
+
+			List<DDMFormField> ddmFormFields = ddmForm.getDDMFormFields();
+
+			DDMFormField ddmFormField = ddmFormFields.stream(
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMFormValuesMergerImpl.java]: Same thing , if it's being used in only one place we should scope it better.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormFieldTemplateContextFactory.java [hunk 2]
```diff
@@ -613,28 +615,32 @@ protected void setDDMFormFieldTemplateContextValueLocalizableValue(
 		DDMFormField ddmFormField = _ddmFormFieldsMap.get(
 			ddmFormFieldValue.getName());
 
-		DDMFormFieldValueAccessor<?> ddmFormFieldValueAccessor =
+		DDMFormFieldValueAccessor<?> ddmFormFieldValueAccessorValue =
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormFieldTemplateContextFactory.java]: Sorry, i might have said you to change this wrongly. This should be **ddmFormFieldValueAccessor** .

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormFieldTemplateContextFactory.java [hunk 3]
```diff
@@ -613,28 +615,32 @@ protected void setDDMFormFieldTemplateContextValueLocalizableValue(
...
-			}
-			else {
-				localizedValue.put(
-					languageId,
-					ddmFormFieldValueAccessor.getValue(
-						ddmFormFieldValue, availableLocale));
+			Object localizedValue = value.getString(availableLocale);
+
+			if (ddmFormFieldValueAccessorValue != null) {
+				Object valueDDMFormFieldValueAccessor =
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormFieldTemplateContextFactory.java]: this sone should be ddmFormFieldValueAccessorValue

**Pull request closed without merging**